cmd/juju/commands/upgradecontroller.go

```diff
@@ -133,6 +138,47 @@ func (c *upgradeControllerCommand) Run(ctx *cmd.Context) (err error) {
 	return c.upgradeIAASController(ctx)
 }
 
+// fetchStreamsVersions returns simplestreams agent metadata
+// for the specified stream. timeout ensures we don't block forever.
+func fetchStreamsVersions(
+	client toolsAPI, majorVersion int, stream string, timeout time.Duration,
+) (tools.List, error) {
+	// Use a go routine so we can timeout.
+	result := make(chan tools.List, 1)
+	errChan := make(chan error, 1)
+	go func() {
+		findResult, err := client.FindTools(majorVersion, -1, "", "", stream)
+		if err == nil {
+			if findResult.Error != nil {
+				err = findResult.Error
+				// We need to deal with older controllers.
+				if strings.HasSuffix(findResult.Error.Message, "not valid") {
+					err = errors.NotValidf("finding stream data for this model")
+				}
+			}
+		}
+		if err != nil {
+			errChan <- err
+		} else {
+			result <- findResult.List
+		}
+	}()
+
+	var resultList tools.List
```

Comment:
not sure why this is declared here, we don't do anything with it except return

Reply:
was left over from older version of the code, fixed